scanners/ios/project.go

```diff
@@ -0,0 +1,176 @@
+	containerToSchemes[p.project.Path] = projectSchemes
+
+	return containerToSchemes, nil
+}
+
+func (p projectContainer) projects() ([]xcodeproject.XcodeProj, []string, error) {
+	return []xcodeproject.XcodeProj{p.project}, []string{}, nil
+}
+
+func (p projectContainer) projectPaths() ([]string, error) {
+	return []string{p.project.Path}, nil
+}
+
+// workspaceContainer ...
+type workspaceContainer struct {
+	workspace xcworkspace.Workspace
+}
+
+func newWorkspace(path string) (workspaceContainer, error) {
+	if !xcworkspace.IsWorkspace(path) {
+		return workspaceContainer{}, fmt.Errorf("path (%s) is not a Workspace", path)
+	}
+
+	workspace, err := xcworkspace.Open(path)
+	if err != nil {
+		return workspaceContainer{}, fmt.Errorf("failed to open Workspace (%s): %v", path, err)
+	}
+
+	return workspaceContainer{
+		workspace: workspace,
+	}, nil
+}
+
+func (w workspaceContainer) isWorkspace() bool {
+	return true
+}
+
+func (w workspaceContainer) path() string {
+	return w.workspace.Path
+}
+
+func (w workspaceContainer) schemes() (map[string][]xcscheme.Scheme, error) {
+	containerToSchemes, err := w.workspace.Schemes()
+	if err != nil {
+		return nil, fmt.Errorf("failed to list Schemes in Workspace (%s): %v", w.workspace.Path, err)
+	}
+
+	return containerToSchemes, nil
+}
+
+func (w workspaceContainer) projects() ([]xcodeproject.XcodeProj, []string, error) {
+	projPaths, err := w.workspace.ProjectFileLocations()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var projects []xcodeproject.XcodeProj
+	var missingProjects []string
+	for _, projPath := range projPaths {
+		if exist, err := pathutil.IsPathExists(projPath); err != nil {
+			return nil, nil, fmt.Errorf("failed to list Projects in the Workspace (%s), can not check if path (%s) exists: %v", w.workspace.Path, projPath, err)
+		} else if !exist {
+			missingProjects = append(missingProjects, projPath)
+			continue
+		}
+
+		project, err := xcodeproject.Open(projPath)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to open Project (%s) in the Workspace (%s): %v", projPath, w.workspace.Path, err)
+		}
+
+		projects = append(projects, project)
+	}
+
+	return projects, missingProjects, nil
+}
+
+func (w workspaceContainer) projectPaths() ([]string, error) {
+	return w.workspace.ProjectFileLocations()
+}
+
+func numberOfSharedSchemes(containerToSchemes map[string][]xcscheme.Scheme) int {
+	var count int
+	for _, schemes := range containerToSchemes {
+		for _, scheme := range schemes {
+			if scheme.IsShared {
+				count++
+			}
+		}
+	}
+
+	return count
+}
+
+func pathRelativeToWorkspace(project, workspace string) string {
+	parentDir, _ := filepath.Split(workspace)
+	relPath, err := filepath.Rel(filepath.Join(parentDir), project)
+	if err != nil {
+		log.Warnf("%s", err)
+		return project
+	}
+
+	return relPath
+}
+
+func printSchemes(includeUserSchemes bool, containerToSchemes map[string][]xcscheme.Scheme, containerPath string) {
```

Comment:
it seems to be unused

Reply:
this means `pathRelativeToWorkspace` has a single usage, and it is used for logging, do we need this? can not we log the absolute path and spare some lines of code?

Reply:
removed completely.